nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -76,19 +148,129 @@ def __init__(self):
         self.left_arm_home_joint_values = CFG["left_arm_home"]["joints"]
         self.right_arm_home_joint_values = CFG["right_arm_home"]["joints"]
 
-    def cmd_right_arm(self, joint_values, blocking=True):
-        goal = joint_goal(joint_values, self.right_arm_joint_names)
-        self.right_arm.send_goal(goal)
-        if blocking:
-            return self.right_arm.wait_for_result()  # TODO: action_server.get_result()
-        return True
+        self._right_joint_states = None
+        self._left_joint_states = None
+        self._joint_tolerance = joint_tolerance
 
-    def cmd_left_arm(self, joint_values, blocking=True):
-        goal = joint_goal(joint_values, self.left_arm_joint_names)
-        self.left_arm.send_goal(goal)
-        if blocking:
-            return self.left_arm.wait_for_result()  # TODO: action_server.get_result()
-        return True
+    def update_joint_states(self):
```

Comment:
Why not use a subscriber to always get the most recent data, then store it and use it as necessary, rather than having a function that only listens for 1 message

Reply:
I believe doing it this way increases performance (and possibly reduces network traffic?), espicially when the manipulation task isn't running. The update is only needed before a control command is sent so it doesn't really make sense to me to continiously update